hypershift-operator/controllers/nodepool/conditions.go

```diff
@@ -852,3 +855,92 @@ func (r NodePoolReconciler) validPlatformConfigCondition(ctx context.Context, no
 	SetStatusCondition(&nodePool.Status.Conditions, *condition)
 	return nil, nil
 }
+
+func (r *NodePoolReconciler) supportedVersionSkewCondition(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster) (*ctrl.Result, error) {
+	if hcluster.Status.Version == nil {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
```

Comment:
nit: The code might be clearer if there was an inner function that returns (bool, error) and the condition is constructed once based on that, instead of constructing the condition on every error condition.